### DISCUSSION TITLE
Running devtools in production

### DISCUSSION THREAD

**Discussion by rrva** [urql-graphql/urql#2822] [Q&A] [answered]
Is the urql devtools exchange suitable to run in production? Should we toggle it on with a query parameter to our app or similar?
How do people use devtools on production sites?

**Comment by JoviDeCroock** [comment]
It currently does not run in production https://github.com/urql-graphql/urql-devtools-exchange/blob/master/src/exchange.ts#L152

**Comment by kitten** [comment, marked as answer]
Currently, the Devtools exchange has a hardcoded switch that disables it and replaces it with a noop in production: https://github.com/urql-graphql/urql-devtools-exchange/blob/fb4a2539d8bf442ec4d37425674f1700524d85dc/src/exchange.ts#L152-L154

This means that you technically won't be able to use it in production. Additionally, the internal events in `@urql/core` (and related packages) it relies on will not be enabled in production, so it won't actually show much if you do get it enabled.

We're looking back into whether we want to change this, however, its overhead is simply a little too large right now to change this. So, while we'll revisit this, we likely won't make it "usable" in production, even if we change how the exchange behaves there
